### DISCUSSION TITLE
Best AI models

### DISCUSSION THREAD

**Discussion by lordraiden** [immich-app/immich#5230] [Q&A]
What are the best models? If the only important thing is quality and not performance.

Image tagging
https://huggingface.co/models?pipeline_tag=image-classification&sort=trending

Smart search
https://huggingface.co/immich-app

Facial recognition: I assume it is **antelopev2**

If I do a search does the search will use the tags as a reference in addition to the smart search feature to show results?

**Comment by DX37** [comment]
Official documentation suggesting you to use `google/vit-base-patch16-224` for better image tagging.

For smart search I use multilingual `XLM-Roberta-Large-Vit-B-32`, because English is not my first language. Works good, and in characteristics of the model it's even slightly better that default `ViT-B-32__openai`.
![изображение](https://github.com/immich-app/immich/assets/12829428/7afba4ae-24b1-4447-ac17-886d51e49479)

And the facial... yeah, there isn't much options, so `antelopev2` is fine, I guess, but it still mistakes some faces and even animals, lol.

**Reply by lordraiden** [reply]
I use ViT-B-32__openai for smart search but if I use words in spanish I still get extremely similar results than if I do the same search in english.
Are you sure that ViT-B-32__openai  is not multilingual or at least doesn't cover the spanish language?


BTW where I can see the tags? I have this pic, If I search dog I can find it but I don't see where the tags are.
![imagen](https://github.com/immich-app/immich/assets/17175395/dfdfc45d-e8c1-43f5-84af-9a96b046d02a)

**Reply by lordraiden** [reply]
if I search m:perro It can find the pic but if search m:dog it doesn't... very strange

**Reply by DX37** [reply]
Don't know much about tags, to be honest. And not sure about multilingual support for `ViT-B-32__openai` model, because on that website it's not listed in corresponding list. Maybe Spanish is kinda similar to English for this model (I apologize if I'm wrong), but, of course, I can't search in Russian with this model, so I have to use model with known multilingual support.

**Reply by DX37** [reply]
Boy, am I blind... There's `XLM-Roberta-Large-Vit-B-16Plus` available for Immich and I using little bit effective model...

**Reply by lordraiden** [reply]
@DX37 I have tried with that one but I get an error in the logs and search don't work. Does it work for you?

**Reply by DX37** [reply]
Just done running Encode CLIP job - works well.

**Comment by kiwijunglist** [comment]
Imaging tagging model is no longer in the settings, I can only find smart search model.  Am I going blind, or was this removed as a setting?

**Reply by bo0tzz** [reply]
It was removed

**Reply by Ador-able** [reply]
> It was removed

Is it possible to modify the image-classification model? What should I do
